src/LinkedList.ts

```diff
@@ -2,54 +2,52 @@ import { fabric } from "fabric";
 import { Node } from "./Node";
 import { makeLine } from "./Utils";
 import Config = require("./Config");
+import { Variable } from "./Variable";
 
 export class LinkedList {
 
-    private head: Node = null;
+    private globalVars: { [key: string]: Variable };
+    private nodes: Node[];
```

Comment:
Is this really necessary? 

I really don't like the idea of having it here because I want the code to mirror the diagram, and its just one more thing to have to keep track of and update whenever the LL changes.

Reply:
I think it would be good if we can have the option to:
- make dangling parts disappear, or
- keep them visible in the diagram

In order to continue drawing dangling nodes, we need to have a nodes array; otherwise there's no way for us to access those nodes.

Reply:
hmm...since we aren't sure exactly how we are going to implement those features in the future, lets leave it out for now and worry about whether we need it or not later.

Reply:
Another idea: maybe we have a list of just the nodes that don't have any pointers to them currently?

Reply:
You said that the nodes array is "just one more thing to have to keep track of and update whenever the LL changes"; however, I don't think that's true. The only time we'd have to modify the nodes array is when the user creates a new node; there's no need to maintain the nodes in any particular order. Essentially, the array represents all of the nodes that currently exist in heap memory.

Reply:
ah ok I see the source of confusion now: I thought that `nodes` was just supposed to be all the nodes currently in the list, but you were intending it to mean all the existing nodes.